### PR TITLE
[AIRFLOW-4936] Add context manager to set temporary config variables.

### PR DESCRIPTION
**Pull request by jmcarp**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-4936
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

File: tests/test_utils/config.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+# under the License.
+
+import contextlib
+import functools
+
+from airflow.configuration import conf
+
+
+@contextlib.contextmanager
+def conf_vars(overrides):
```
Comment (ashb) [inline tests/test_utils/config.py:26]: See https://issues.apache.org/jira/browse/AIRFLOW-4865. We already have a tests/test_utils/decorators.py that contains a mock_conf_get decorator that we should use everywhere we need to change a config value in a test. 

Could you also re-target against that Jira?

Comment (jmcarp) [reply tests/test_utils/config.py:26]: Thanks, I updated the ticket in the commit message. It looks like the `mock_conf_get` decorator / context manager isn't used anywhere, and handles fewer cases than the decorator I added, so I'm proposing to delete it and just use the new code. I also simplified the proposed new decorator--I didn't know that `contextlib.contextmanager` produces a function that's both a decorator and a context manager.

File: tests/contrib/operators/test_ssh_operator.py [hunk 2]
```diff
@@ -87,15 +86,15 @@ def test_json_command_execution(self):
...
+        with conf_vars({('core', 'enable_xcom_pickling'): 'False'}):
+            ti = TaskInstance(
+                task=task, execution_date=timezone.utcnow())
+            ti.run()
+            self.assertIsNotNone(ti.duration)
+            self.assertEqual(ti.xcom_pull(task_ids='test', key='return_value'),
+                             b64encode(b'airflow').decode('utf-8'))
 
     def test_pickle_command_execution(self):
-        configuration.conf.set("core", "enable_xcom_pickling", "True")
```
Comment (ashb) [inline tests/contrib/operators/test_ssh_operator.py:98]: Use the decorators here too?

(I think I prefer the decorator approach as it means the code is indented less.)

File: tests/test_utils/config.py (hunk 1 above)
Comment (ashb) [reply tests/test_utils/config.py:26]: That was used in one or two places, I guess that got removed, but this new one is nicer!

**Comment by ashb** [comment]
Could you also fix up https://github.com/apache/airflow/blob/18e60f4a29881a6ad6b480296f70d98c9caedbea/tests/executors/test_kubernetes_executor.py#L195-L222

**Comment by ashb** [comment]
And a couple more:

```
$ pt -e "@.*patch.*conf"
./tests/models/test_connection.py
39:    @patch('airflow.configuration.conf.get')
51:    @patch('airflow.configuration.conf.get')
61:    @patch('airflow.configuration.conf.get')

./tests/models/test_variable.py
36:    @patch('airflow.configuration.conf.get')
48:    @patch('airflow.configuration.conf.get')
60:    @patch('airflow.configuration.conf.get')

./tests/cli/test_worker_initialisation.py
48:    @mock.patch('airflow.configuration.conf.getboolean')
60:    @mock.patch('airflow.configuration.conf.getboolean')
```

**Comment by ashb** [comment]
Good work btw @jmcarp -- this has been on my wish list for a while! 👍

**Comment by jmcarp** [comment]
Ready for another look @ashb.

**Review by ashb — APPROVED** [review]

**Pull request merged**